src/Console/Command/NewVersionChecker.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Console\Command;
+
+/**
+ * @internal
+ */
+class NewVersionChecker
+{
+    /**
+     * @var int[]
+     */
+    private $currentVersion;
+
+    /**
+     * @var null|array<int[]>
+     */
+    private $availableVersions;
+
+    /**
+     * @param string $currentVersion
+     */
+    public function __construct($currentVersion)
+    {
+        $this->currentVersion = $this->parseVersion($currentVersion);
+    }
+
+    /**
+     * Returns the tag of the latest version if newer than the current one.
+     *
+     * @return null|string
+     */
+    public function getLatestVersion()
+    {
+        $this->retrieveAvailableVersions();
+
+        return $this->returnIfNewer($this->availableVersions[0]);
+    }
+
+    /**
+     * Returns the tag of the latest minor/patch version if newer than the current one.
+     *
+     * @return null|string
+     */
+    public function getLatestVersionOfCurrentMajor()
+    {
+        $this->retrieveAvailableVersions();
+
+        foreach ($this->availableVersions as $availableVersion) {
+            if ($availableVersion[0] === $this->currentVersion[0]) {
+                return $this->returnIfNewer($availableVersion);
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * @param int[] $newVersion
+     *
+     * @return null|string
+     */
+    private function returnIfNewer($newVersion)
+    {
+        if (1 === $this->compareVersions($newVersion, $this->currentVersion)) {
+            return 'v'.implode('.', $newVersion);
+        }
+
+        return null;
+    }
+
+    private function retrieveAvailableVersions()
+    {
+        if (null !== $this->availableVersions) {
+            return;
+        }
+
+        $url = 'https://api.github.com/repos/FriendsOfPHP/PHP-CS-Fixer/tags';
+
+        $result = @file_get_contents(
+            $url,
+            false,
+            stream_context_set_default(array(
```

Comment:
This is process wide right? we should set back .. although edge case..
Would a move to Guzzle or `curl` be hard or have serious side effects for us? Asking because both would provide more details and control when something does not work as expected, like getting the response code, dealing with timeouts etc.

Reply:
Replaced `stream_context_set_default()` with `stream_context_create()`, I think this is the only place where we use this kind of context.

I'm 👍 for moving to curl or Guzzle but I think it's out of the scope of this PR.

Reply:
:-1: for guzzle, it's extra big library we don't need